odxtools/decodestate.py

```diff
@@ -103,7 +103,9 @@ def extract_atomic_value(
 
         # Deal with raw byte fields, ...
         if base_data_type == DataType.A_BYTEFIELD:
-            odxassert(base_type_encoding in (None, Encoding.NONE, Encoding.BCD_P, Encoding.BCD_UP))
+            odxassert(
+                base_type_encoding in (None, Encoding.NONE, Encoding.BCD_P, Encoding.BCD_UP),
+                f"Illegal encoding '{base_type_encoding}' for A_BYTEFIELD")
```

Comment:
@andlaus BCD_P and BCD_UP are valid for A_BYTEFIELD

See table 8 of the specification, page 99

Reply:
yes, that's why we assert it. (the condition must be true for assertations ;))